azure-functions/acbs-function/mappings/facility/facility-loan.js

```diff
@@ -22,6 +22,35 @@ const date = require('../../helpers/date');
 const CONSTANTS = require('../../constants');
 const getDealSubmissionDate = require('../deal/helpers/get-deal-submission-date');
 
+/**
+ * Maps the facility loan record for a given deal and facility.
+ *
+ * This function performs the following operations:
+ * 1. Extracts the issue date and guarantee expiry date from the facility's guarantee dates.
+ * 2. Retrieves the currency of the facility, defaulting to a constant if not provided.
+ * 3. Calculates the maximum liability and other financial details using helper functions.
+ * 4. Constructs and returns the facility loan record object.
+ * 5. Adds additional fields if the facility is not in GBP.
+ * 6. Adds specific fields for BSS/EWCS deals.
+ *
+ * @param {Object} deal - The deal object containing deal details.
+ * @param {Object} deal.dealSnapshot - The snapshot of the deal details.
```

Comment:
do we want to commonise these object definitions to make the code more maintainable and easier to read?